### PR TITLE
Complete transport manager unit tests

### PR DESCRIPTION
**Pull request by VVeremjova**
@malirod, @AByzhynar, @AMikhniev Please review

### CODE REVIEW

File: src/components/transport_manager/test/transport_manager_impl_test.cc [hunk 1]
```diff
@@ -49,235 +54,276 @@ using ::testing::Return;
...
 
-  EXPECT_CALL(*tm_listener, OnConnectionClosed(application_id));
+  void UninitializeTM() {
+    EXPECT_CALL(*mock_adapter_, Terminate());
+    ASSERT_EQ(E_SUCCESS, tm_.Stop());
+  }
 
-  tm.TestHandle(test_event);
-}
+  MockTransportManagerImpl tm_;
```
Comment (AByzhynar) [inline src/components/transport_manager/test/transport_manager_impl_test.cc]: @VVeremjova I suggest to use more clear name (like `mock_transport_manager_`) as you did in line below.

**Comment by AMikhniev** [comment]
Reviewed

File: src/components/transport_manager/test/transport_manager_impl_test.cc [hunk 2]
```diff
@@ -417,14 +414,15 @@ TEST_F(TransportManagerImplTest, Disconnect_DisconnectionFailed) {
   EXPECT_CALL(*mock_adapter_, Disconnect(mac_address_, application_id_))
       .WillOnce(Return(TransportAdapter::FAIL));
   // Assert
-  EXPECT_EQ(E_INTERNAL_ERROR, tm_.Disconnect(connection_key_));
+  // Even with fail, we get Success
```
Comment (AByzhynar) [inline src/components/transport_manager/test/transport_manager_impl_test.cc]: @VVeremjova May be I don't fully understand the logic but comment looks strange for me.

**Comment by AByzhynar** [comment]
Reviewed

File: src/components/transport_manager/test/transport_manager_impl_test.cc (hunk 2 above)
Comment (VVeremjova) [inline src/components/transport_manager/test/transport_manager_impl_test.cc]: @AByzhynar This comment explains why with TransportAdapter::FAIL we expect Success - that is the current logic in GENIVI

**Comment by malirod** [comment]
Reviewed

**Pull request merged**